addons/event/models/event_event.py

```diff
@@ -771,3 +773,27 @@ def _gc_mark_events_done(self):
         ])
         if ended_events:
             ended_events.action_set_done()
+
+    def _get_event_timeframe_string(self):
+        self.ensure_one()
+        start_datetime = format_datetime(self.env, self.date_begin, self.date_tz, "short")
+        if self.is_one_day:
+            end_datetime = format_time(self.env, self.date_end, self.date_tz, "short")
+        else:
+            end_datetime = format_datetime(self.env, self.date_end, self.date_tz, "short")
+        return _("%(start_date)s to %(end_date)s", start_date=start_datetime, end_date=end_datetime)
+
+    def _get_event_print_details(self):
+        self.ensure_one()
+        return {
+            'name': self.name,
+            'badge_image': self.badge_image,
+            'timeframe': self._get_event_timeframe_string(),
+            'address': self.address_id.name if self.address_id else None,
+            'logo': self.company_id.logo,
+            'sponsor_text': self._get_printing_sponsor_text()
+        }
+
+    def _get_printing_sponsor_text(self):
+        sponsor_text = self.env['ir.config_parameter'].sudo().get_param('event.badge_printing_sponsor_text')
+        return sponsor_text or "Powered by Odoo"
```

Comment:
Minor note
Not sure if missing translation is on purpose, don't think it's a big deal but as we translate the time :shrug: 

Also globally we could make sure badges are printed in the language of the event if any is set (?), so if Francois LeFranc prints a ticket on his account for the OXP the time is still in english :shrug: